lib/streams/stream.ts

```diff
@@ -8,13 +8,17 @@ export interface StreamOptions {
 }
 
 export default class Stream implements StreamInterface {
-  private options: StreamOptions;
+  private init: Function;
+  private queue: QueueInterface;
+  private consume: Boolean;
   private _process: Function;
   private targets: Array<StreamInterface> = [];
   private refCount: number = 0;
   private populateCallbacks: Array<Function> = [];
   constructor (options: StreamOptions = {}) {
```

Comment:
I don't know if you want to call this with a property bag e.g. `options`. This is the short-lived map problem in V8 https://docs.google.com/document/d/1VEeXn6BfKmVkYpfMuxNs3otLLNPIxIVA1cLoSwGMIxI/edit